core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -249,6 +266,15 @@ private ClientTransport obtainActiveTransport() {
         // start() must return before we set activeTransport, since activeTransport is accessed
         // without a lock.
         activeTransport = newActiveTransport;
+        backoff = null;
```

Comment:
Rather than doing this, could we give `RetryPolicy` a `reset` method which puts it back to it's original settings? That way `ChannelImpl` could be agnostic as to what kind of `RetryPolicy` it's using.